### PR TITLE
Limit number of time series in explore

### PR DESCRIPTION
**Pull request by peterholmberg**
# What 
Added a limit to how many time series we render in graph in explore. Also added an option to render all time series. 

# Why 
To reduce cpu load when rendering graph.

### CODE REVIEW

File: public/sass/pages/_explore.scss [hunk 1]
```diff
@@ -55,6 +55,25 @@
     margin-top: 2 * $panel-margin;
   }
 
+  .time-series-disclaimer {
+    width: 300px;
+    margin: 10px auto;
+    padding: 10px 0;
+    border-radius: 4px;
+    text-align: center;
+    background-color: #212124;
```
Comment (torkelo) [inline public/sass/pages/_explore.scss]: should use variables (both for consistency and to support both dark and light themes)

File: public/sass/pages/_explore.scss [hunk 2]
```diff
@@ -55,6 +55,25 @@
     margin-top: 2 * $panel-margin;
   }
 
+  .time-series-disclaimer {
+    width: 300px;
```
Comment (davkal) [inline public/sass/pages/_explore.scss]: I'd make that `min-width`, and then give it an `em` value for scaling.

File: public/sass/pages/_explore.scss [hunk 3]
```diff
@@ -55,6 +55,25 @@
...
+    margin: 10px auto;
```
Comment (davkal) [inline public/sass/pages/_explore.scss]: Please use variables here and for all values below as well. Check out the other classes in this file. For small values just divide a base variable.

**Review by davkal — CHANGES_REQUESTED** [review]
Excellent work. Just small nits.

File: public/sass/pages/_explore.scss (hunk 2 above)
Comment (torkelo) [reply public/sass/pages/_explore.scss]: em value? We don't use em values anywhere. We used to use mostly rems but are moving from rems to pixels as the argument to use rems are mostly gone now that browser handle the scaling correctly for both rems & pixels and defining size in pixels is so much easier and accurate (create less floating point pixels and other misalignments that rems can cause)

File: public/app/features/explore/Graph.tsx [hunk 4]
```diff
@@ -124,18 +147,32 @@ class Graph extends Component<any, any> {
...
           <div className="muted m-a-1">The queries returned no time series to graph.</div>
         </div>
       );
     }
     return (
-      <div className="panel-container">
-        <div id={this.props.id} className="explore-graph" style={{ height }} />
-        <Legend data={data} />
+      <div>
+        {this.props.data.length > 20 &&
```
Comment (davkal) [inline public/app/features/explore/Graph.tsx]: Need to use constant here too!

File: public/sass/pages/_explore.scss [hunk 5]
```diff
@@ -55,6 +55,25 @@
     margin-top: 2 * $panel-margin;
   }
 
+  .time-series-disclaimer {
+    width: 300px;
+    margin: $panel-margin auto;
+    padding: 10px 0;
```
Comment (davkal) [inline public/sass/pages/_explore.scss]: No variable made sense here?

**Review by davkal — CHANGES_REQUESTED** [review]
Last fix needed: use constant in render function.

File: public/sass/pages/_explore.scss (hunk 5 above)
Comment (peterholmberg) [reply public/sass/pages/_explore.scss]: Not to me, felt weird to use a margin-variable on padding. And the padding variable wouldn't match.

File: public/app/features/explore/Graph.tsx [hunk 6]
```diff
@@ -124,18 +147,32 @@ class Graph extends Component<any, any> {
...
     return (
-      <div className="panel-container">
-        <div id={this.props.id} className="explore-graph" style={{ height }} />
-        <Legend data={data} />
+      <div>
+        {this.props.data.length > MAX_NUMBER_OF_TIME_SERIES &&
+          !this.state.showAllTimeSeries && (
+            <div className="time-series-disclaimer">
+              <i className="fa fa-fw fa-warning disclaimer-icon" />
+              Showing only 20 time series.{' '}
```
Comment (davkal) [inline public/app/features/explore/Graph.tsx]: Ooops.

**Review by davkal — CHANGES_REQUESTED** [review]
So close!

File: public/app/features/explore/Graph.tsx (hunk 6 above)
Comment (peterholmberg) [reply public/app/features/explore/Graph.tsx]: Haha 😂

**Review by davkal — APPROVED** [review]

**Comment by davkal** [comment]
All good now!

**Pull request merged**